extensions/amp-pinterest/0.1/amp-pinterest.js

```diff
@@ -0,0 +1,574 @@
+                  'className': 'amp-pinterest-embed-pin-stats-repins',
+                  'textContent': String(p.repin_count)
+                }});
+                stats.appendChild(repinCount);
+              }
+
+              if (p.like_count) {
+                let likeCount = make({'SPAN': {
+                  'className': 'amp-pinterest-embed-pin-stats-likes',
+                  'textContent': String(p.like_count)
+                }});
+                stats.appendChild(likeCount);
+              }
+              text.appendChild(stats);
+            }
+
+            // pinner
+            if (p.pinner) {
+
+              let pinner = make({'SPAN':{
+                'className': 'amp-pinterest-embed-pin-text-block' +
+                  ' amp-pinterest-embed-pin-pinner'
+              }});
+
+              // avatar
+              pinner.appendChild(make({'IMG':{
+                'className': 'amp-pinterest-embed-pin-pinner-avatar',
+                'alt': filter(p.pinner.full_name),
+                'title': filter(p.pinner.full_name),
+                'src': p.pinner.image_small_url,
+                'data-pin-href': p.pinner.profile_url
+              }}));
+
+              // name
+              pinner.appendChild(make({'SPAN': {
+                'className': 'amp-pinterest-embed-pin-pinner-name',
+                'textContent': filter(p.pinner.full_name),
+                'data-pin-href': p.pinner.profile_url
+              }}));
+
+              // board
+              pinner.appendChild(make({'SPAN': {
+                'className': 'amp-pinterest-embed-pin-board-name',
+                'textContent': filter(p.board.name),
+                'data-pin-href': 'https://www.pinterest.com/' + p.board.url
+              }}));
+
+              text.appendChild(pinner);
+            }
+
+            container.appendChild(text);
+            structure.appendChild(container);
+
+            // listen for clicks
+            structure.addEventListener('click', function (e) {
+              let el = e.target;
+              let logMe = el.getAttribute('data-pin-log');
+              if (logMe) {
+                log('&type=' + logMe);
+              }
+              let href = el.getAttribute('data-pin-href');
+              if (href) {
+                let popThis = el.getAttribute('data-pin-pop') || false;
+                pop(href, popThis);
+              }
+              e.preventDefault();
+            });
+
+            // fill it
+            that.applyFillContent(structure);
+            // append it
+            that.element.appendChild(structure);
+            // done
+            return loadPromise(structure);
+
+          }
+        };
+
+
+        let query = 'https://widgets.pinterest.com/v3/pidgets/pins/info/' +
+          '?pin_ids=' + pinId +
+          '&sub=www&base_scheme=https';
+        return call(query).then((r) => {
+          renderPin(r);
+          return loadPromise(structure);
+        });
+
+      break;
```

Comment:
Having a hard time with this. Although all my buttons and widgets are rendering, applyFillContent is throwing errors. I'm sure it's something I don't understand about how promises work.